lightning/src/offers/flow.rs

```diff
@@ -449,61 +449,107 @@ where
 		}
 	}
 
+	fn create_offer_builder_intern<ES: Deref, PF>(
+		&self, entropy_source: ES, make_paths: PF,
+	) -> Result<(OfferBuilder<DerivedMetadata, secp256k1::All>, Nonce), Bolt12SemanticError>
+	where
+		ES::Target: EntropySource,
+		PF: FnOnce(
+			PublicKey,
+			MessageContext,
+			&secp256k1::Secp256k1<secp256k1::All>,
+		) -> Result<Vec<BlindedMessagePath>, Bolt12SemanticError>,
```

Comment:
Can this return an `Iterator` so we don't need to `collect` in two instances below?

Reply:
Makes sense! Updated in [pr3246.23](https://github.com/shaavan/rust-lightning/commits/pr3246.23) to return iterator, instead of `vec`. Thanks!